host-provision.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+echo Command used:
+echo "$0 $@"
+echo
+
+shopt -s nocasematch
+
+# Check if we're using the Mac stock getopt and fail if true
+out=`getopt -T`
+if [ $? != 4 ]; then
+  echo -e "Your getopt does not support long parameters, possibly you're on a Mac, if so please install gnu-getopt with brew"
+  echo -e "\thttps://brewformulas.org/Gnu-getopt"
+  exit
+fi
+
+# set -x
+GETOPT_MANDATORY="inventory-file:"
+# GETOPT_OPTIONAL="ora-role-separation:,ora-disk-mgmt:,ora-swlib-path:,ora-staging:,ora-asm-disks:,ora-data-mounts:,help"
+GETOPT_OPTIONAL="hostprovision-ansible-user:,help"
+GETOPT_LONG="${GETOPT_MANDATORY},${GETOPT_OPTIONAL}"
+GETOPT_SHORT="h"
+
+
+INVENTORY_FILE="${INVENTORY_FILE:-./inventory_files/inventory}"
+
+
+options=$(getopt --longoptions "$GETOPT_LONG" --options "$GETOPT_SHORT" -- "$@")
+
+[ $? -eq 0 ] || {
+       echo "Invalid options provided: $*"
+       exit 1
+}
+
+# echo "PARSED COMMAND LINE FLAGS: $options"
+
+eval set -- "$options"
+
+while true; do
+    case "$1" in
+    --inventory-file)
```

Comment:
I'd really really like to avoid having to create yet another configuration file manually.  Can we please follow the existing logic in install-oracle.sh, taking parameters as required for host IPs (or a rac config file), and generating an inventory file from that?

Reply:
I see your point and agree with it.

My thoughts on this:
* For the `install-oracle.sh`, inventory variables and their parsing are important and we feed `--cluster-type RAC` and  `--cluster-config mntrl_cluster.json`.
    * the json file is dense with necessary info consumed by: `install-oracle.sh`.
* For the current `host-provision.sh`, we basically need the host IPs to connect to and there is no need to build an inventory (with the many host/group vars) at this stage of the host
   * consequentially, will explore the passing of IPs into `host-provision.sh` via cmd line and avoid having to create by hand another config file

Guess, this will satisfy the condition you pointed out of not having to hand-code and feed another config file...while, at the same time, keeping the overall design/flow simple.

Reply:
This has been done.